forem/forem
# Only show the "Targeted Tags Field" when the selected placement area is "Below the comment section"

## What type of PR is this? (check all applicable)


## Description

The "Targeted tags" field should only be shown when the selected Placement Area is "Below the comment section". Hence we show/hide field based on the placement area value. 

## Related Tickets & Document
- Related Issue #https://github.com/forem/forem/issues/18519
- Closes #

## QA Instructions, Screenshots, Recordings

A Loom video to demo: 
https://www.loom.com/share/8cbd8175c8f04cd980c97c6b84c4d5a5

QA Instructions: 

Create a New Display Ad:
1. Toggle the feature flag by running `FeatureFlag.enable(:display_ad_tags)` in the rails console.
2. Create a new Display Ad by going to `http://localhost:3000/admin/customization/display_ads/new`
3. Start filling out the information for the Display Ad. When you select a Placement Area of "Below the comment section", then you should see the Targeted Tags field appear. 
4. Try saving the Display Ad and ensure the tags are still saving by running `DisplayAd.last.tags` in the rails console.

Go back and edit the Display Ad that you created:
1. Edit the last created Display Ad by changing the placement area to any other value from the dropdown. 
2. You should see the Targeted Tags field disappear on the UI.
3. Now save the Display Ad. 
4. You should run `DisplayAd.last.tags` in the rails console and note the tags should be an empty array. 

Play around with error conditions on creation as well to ensure that the Targeted Tag(s) are behaving as outlined in the [issue](https://github.com/forem/forem/issues/18519).


### UI accessibility concerns?

There probably are some accessibility concerns here, but I need to do some research on how to solve them. I'm going to try to tackle this at the end of the feature.  

## Added/updated tests?

- [x] Yes, but I've skipped them on the build file however they work locally. Rationale explained as a comment in the file. 

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

      [Storybook](https://storybook.forem.com/) (for Crayons components)
      updated. I have filled out the
      [Changes Requested](https://github.com/forem/admin-docs/issues/new?assignees=&labels=&template=changes_requested.md)
      issue template so Community Success can help update the Admin Docs
      appropriately.
      [`CHANGELOG.md`](https://github.com/forem/forem/tree/main/CHANGELOG.md)
      or in a [forem.dev](http://forem.dev) post
      replace this line with details on why this change doesn't need to be
      shared_

## [optional] Are there any post deployment tasks we need to perform?

## [optional] What gif best describes this PR or how it makes you feel?

![alt_text](https://media.giphy.com/media/xT0xemn9kAM4F52NFK/giphy.gif)
